ChustaSoft.Tools.DBAccess.MongoDb/Repository.cs

```diff
@@ -1,79 +1,102 @@
-        where TEntity : class
+        where TEntity : class, IKeyable<TKey>
     {
+        private readonly IMongoCollection<TEntity> _dbSet;
 
         public IQueryable<TEntity> Query => throw new NotImplementedException();
 
 
         public Repository(IMongoContext context) 
             : base(context)
-        { }
-
-
+        {
+            _dbSet = context.GetCollection<TEntity>();
+        }
 
         public TEntity GetSingle(TKey id)
         {
-            throw new NotImplementedException();
+            var idFilter = CreateIdFilter(id);
+
+            return _dbSet
+                .Find(idFilter)
+                .FirstOrDefault();
         }
 
         public TEntity GetSingle(Action<ISingleResultSearchParametersBuilder<TEntity>> searchCriteria)
         {
-            throw new NotImplementedException();
+            var searchParams = MongoSearchParametersBuilder<TEntity, MongoSearchParameters<TEntity>>.GetParametersFromCriteria(searchCriteria);
+
+            return _dbSet.Find(searchParams.Filter).FirstOrDefault();
         }
 
         public IEnumerable<TEntity> GetMultiple(Action<ISearchParametersBuilder<TEntity>> searchCriteria)
         {
-            throw new NotImplementedException();
+            var searchParams = MongoSearchParametersBuilder<TEntity, MongoSearchParameters<TEntity>>.GetParametersFromCriteria(searchCriteria);
+
+            return _dbSet.Find(searchParams.Filter).ToList();
         }
 
         public void Insert(TEntity entity)
         {
-            throw new NotImplementedException();
+            _dbSet.InsertOne(entity);
         }
 
         public void Insert(IEnumerable<TEntity> entities)
         {
-            throw new NotImplementedException();
+            _dbSet.InsertMany(entities);
         }
 
         public void Update(TEntity entity)
         {
-            throw new NotImplementedException();
+            var idFilter = CreateIdFilter(entity.Id);
+
+            _dbSet.ReplaceOne(idFilter, entity);
         }
 
         public void Update(IEnumerable<TEntity> entities)
         {
-            throw new NotImplementedException();
+            foreach (var entity in entities)
+            {
+                Update(entity);
+            }
         }
 
         public void Delete(TKey id)
         {
-            throw new NotImplementedException();
+            var idFilter = CreateIdFilter(id);
+
+            _dbSet.DeleteOne(idFilter);
         }
 
         public void Delete(TEntity entity)
         {
-            throw new NotImplementedException();
+            var idFilter = CreateIdFilter(entity.Id);
+
+            _dbSet.DeleteOne(idFilter);
         }
 
 
         protected override IQueryable<TEntity> GetQueryable()
         {
-            throw new NotImplementedException();
+            return _dbSet.AsQueryable();
         }
 
+        private FilterDefinition<TEntity> CreateIdFilter<T>(T id)
```

Comment:
In order to simpmy naming: Rename to GetFilter or CreateFilter, param id is already defining the context

Reply:
[Corrected in linked commit](https://github.com/ChustaSoft/DBAccess/pull/22/commits/8654299179af4b573a1b7af2a9357613e22fb227)